django/interface/views.py

```diff
@@ -16,8 +18,94 @@ def statistics(request):
 
 
 def estimate(request):
-    """View of interface estimate"""
-    return render(request, "interface/estimate.html")
+    """View of interface estimate
+    This function calculates the average stay of pets in the shelter based on the
+    get parameters.
+    If there are not enough pets to estimate the range is increased until there are
+    NUMBER_OF_PETS_NEEDED_FOR_ESTIMATE pets. Otherwise error is displayed.
+    If an estimate was calculated some example pets are displayed."""
+
+    context = {}
+
+    breeds = Pet.objects.values("breed").exclude(breed="").distinct()
+    context["breeds"] = breeds
+
+    breed = request.GET.get("breed")
+    age = request.GET.get("age")
+    weight = request.GET.get("weight")
+    gender = request.GET.get("sex")
+
+    context["breed"] = breed
+    context["age"] = age
+    context["weight"] = weight
+    context["gender"] = gender
+
+    if gender is not None:
+        gender = gender.lower()
+
+    min_age = None
+    max_age = None
+    min_weight = None
+    max_weight = None
+
+    pets = Pet.objects.all()
+
+    if not (
+        (breed is None) and (age is None) and (weight is None) and (gender is None)
```

Comment:
This is too complex if logic. Please simplify

Reply:
This logic seem very straightforward. Could you elaborate what do you mean?